### PR TITLE
Fix Api Test Failure

### PR DESCRIPTION
**Pull request by sudhabisht**
Some 4.6 file changes were not included in master

### CODE REVIEW

File: CRM/Member/BAO/Membership.php [hunk 1]
```diff
@@ -2304,12 +2304,15 @@ public static function linkMembershipPayment($membership, $membershipContributio
    * @return array
    */
   public static function createOrRenewMembership($membershipParams, $contactID, $customFieldsFormatted, $membershipID, $memType, $isTest, $numTerms, $membershipContribution, &$form) {
+    if (!empty($membershipContribution)) {
+      $pending = ($membershipContribution->contribution_status_id == 2) ? TRUE : FALSE;
```
Comment (yashodha) [inline CRM/Member/BAO/Membership.php]: Why are you hardcoding status id?

**Comment by eileenmcnaughton** [comment]
This code will be heavily re-written in 4.7  -  but if it decreases test fails you may as well merge in the meantime.

Note that we have agreed to move away from setting contribution_status_id on the payment processor to denote whether a payment succeeded or not & in 4.6 will support both contribution_status_is =1 & payment_status_id=1, in 4.7 we are going to remove the handling for contribution_status_id & only support payment_status_id=1.

In 4.6 contribution_status_id is set to 1 by processors like IATS & eway to indicate the first payment in a recurring series has taken place. (there are no cases where it would be set to 2 at the moment).

in 4.7 the doPayment function will ensure payment_status_id is always set to either completed (1) or pending (2) in the result array.

**Comment by monishdeb** [comment]
@eileenmcnaughton actually the test build failure arises due to merge-forwarding few of the changes of https://issues.civicrm.org/jira/browse/CRM-16520 4.6 to master. I agree it would rather be handled through payment's status rather then contribution status. But for the time being (after the minor change 'removing hard-coded' status value) we can merge this PR to clear the test failures.

**Comment by eileenmcnaughton** [comment]
I agree & will merge

**Comment by sudhabisht** [comment]
Removed hardcode value. PR #6083

**Pull request merged**